cpp/ql/src/Best Practices/Likely Errors/CommaBeforeMisleadingIndentation.qhelp

```diff
@@ -0,0 +1,31 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+
+<overview>
+<p>
+If the expression to the right of a comma operator starts at an earlier column than the expression to the left, then
+this suspicious indentation possibly indicates a logic error, caused by a typo that may escape visual inspection.
+</p>
+<warning>
+This query has medium precision because CodeQL currently does not distinguish between tabs and spaces in whitespace.
+Alerts may therefore flag code that appears readable for one value of tab size but not another.
+</warning>
+</overview>
+
+<recommendation>
+<p>
+Use standard indentation around the comma operator: begin the right-hand-side operand at the same level of
+indentation (column number) as the left-hand-side operand.
+</p>
+<p>
+When it comes to whitespace, either do not mix tabs and spaces, or mix them consistently.
+</p>
+</recommendation>
+
+<example>
+<sample src="CommaBeforeMisleadingIndentation.cpp" />
```

Comment:
We usually expect there to be some introduction or explanation of the example. One possibility is to convert the example into three different example files. Alternatively, something like this might work:

```suggestion
<example>
<p>
This example shows three different ways of writing the same code. The first example contains a comma instead of a semicolon which means that the final line is part of the <code>if</code> statement, even though the indentation suggests that it is intended to be separate. The second example looks different put is functionally the same as the first example. It is more likely that the developer intended to write the third example.
</p>
<sample src="CommaBeforeMisleadingIndentation.cpp" />
```

Reply:
`put` -> `but` in your suggestion :)

Reply:
Thanks Geoffrey 🤦🏻‍♀️